### ISSUE TITLE
Bug - remove fails on VirtualMap

### ISSUE THREAD

**Issue by adamretter** [lacuna/bifurcan#23]
When starting from a `LinearMap` and converting to a `Map`, the subsequent `remove(key)` call on the returned `VirtualMap` does nothing. I think this is probably a bug!

```java
// TEST 1 (FAILS)
final IMap<String, String> linearMap = LinearMap.from(Arrays.asList(new Maps.Entry<>("a", "b")));
assert(linearMap.contains("a"));

final IMap<String, String> map = linearMap.forked();
assert(map.contains("a"));

assert(!map.remove("a").contains("a"));  // this FAILS
```

Interestingly when working the other way around, i.e. from `Map` to `LinearMap` everything works fine, i.e. 
```java
// TEST 2 (OK)
final IMap<String, String> map = Map.from(Arrays.asList(new Maps.Entry<>("a", "b")));
assert(map.contains("a"));

final IMap<String, String> linearMap = map.linear();
assert(linearMap.contains("a"));

assert(!linearMap.remove("a").contains("a"));  // this is OK
```

Even stranger, this one also seems to work, which makes me think the problem is definitely caused by `LinearMap#forked()` returning a `VirtualCollection` (in our first example):
```java
// TEST 3 (OK)
final IMap<String, String> map = Map.from(Arrays.asList(new Maps.Entry<>("a", "b")));
assert(map.contains("a"));

final IMap<String, String> linearMap = map.linear();
assert(linearMap.contains("a"));

final IMap<String, String> map2 = linearMap.forked();

assert(!map2.remove("a").contains("a"));  // this is OK
```

**Comment by ztellman** [comment]
This is resolved in `0.2.0-alpha2`

**Comment by adamretter** [comment]
Thanks @ztellman
